### ISSUE TITLE
Reject starting with multiple threads if `--handle-signals=no`

### ISSUE THREAD

**Issue by vchuravy** [JuliaLang/julia#50278]
Julia GC safepoint is based on a read fault on a protected page. If you start Julia with multiple threads and pass `--handle-signals=no` the signal handler will cause a unhandled segmentation fault.

I don't know if we could install a "partial" segmentation fault handler that only handles safepoints, 
but we should not let users shoot themselves into the foot. An easy solution is to document that multi-threading requires the signal-handlers and to not allow starting with multiple threads when `--handle-signals=no`.

This is of course only a partial fix, since foreign threads may become adopted nontheless.

As seen in the wild: https://discourse.julialang.org/t/floop-threading-and-juliacall-produce-segmentation-fault/100717/11\

x-ref: https://github.com/cjdoris/PythonCall.jl/issues/219

**Comment by vtjnash** [comment]
Or add something like libjsig to our distribution? https://docs.oracle.com/javase/8/docs/technotes/guides/vm/signal-chaining.html

**Comment by vchuravy** [comment]
Yeah that would make cases like https://github.com/JuliaLang/julia/discussions/48378 workable, and maybe better the MPI situation as well.

**Comment by vchuravy** [comment]
`libsig` is GPLv2 licensed as part of the OpenJDK, but the idea is fairly simple for someone to implement.
